test/Microsoft.AspNet.Mvc.Core.Test/ParameterBinding/ModelBindingHelperTest.cs

```diff
@@ -696,6 +696,107 @@ public void GetPropertyName_FieldExpression_Throws()
             Assert.Equal(expectedMessage, exception.Message);
         }
 
+        [Fact]
+        public void ClearValidationStateForModel_EmtpyModelKey()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            ModelStateDictionary dictionary = new ModelStateDictionary();
+            dictionary["MyProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("MyProperty", "MyProperty invalid.");
+            dictionary["IncludedProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("IncludedProperty", "Included Property invalid.");
+            dictionary.AddModelError("IncludedProperty", "Included Property required.");
+            dictionary["ExcludedProperty"] = new ModelState { ValidationState = ModelValidationState.Valid };
+
+            // Act
+            ModelBindingHelper.ClearValidationStateForModel(
+                typeof(MyModel),
+                dictionary,
+                metadataProvider,
+                "");
+
+            // Assert
+            Assert.Equal(0, dictionary["MyProperty"].Errors.Count);
+            Assert.Equal(ModelValidationState.Unvalidated, dictionary["MyProperty"].ValidationState);
+            Assert.Equal(0, dictionary["IncludedProperty"].Errors.Count);
+            Assert.Equal(ModelValidationState.Unvalidated, dictionary["IncludedProperty"].ValidationState);
+            Assert.Equal(0, dictionary["ExcludedProperty"].Errors.Count);
+            Assert.Equal(ModelValidationState.Unvalidated, dictionary["ExcludedProperty"].ValidationState);
+        }
+
+        [Fact]
+        public void ClearValidationStateForCollectionsModel_EmtpyModelKey()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            ModelStateDictionary dictionary = new ModelStateDictionary();
+            dictionary["[0].MyProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("[0].MyProperty", "MyProperty invalid.");
+            dictionary["[0].IncludedProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("[0].IncludedProperty", "Included Property invalid.");
+            dictionary.AddModelError("[0].IncludedProperty", "Included Property required.");
+            dictionary["[0].ExcludedProperty"] = new ModelState { ValidationState = ModelValidationState.Valid };
+
+            dictionary["[1].MyProperty"] = new ModelState { ValidationState = ModelValidationState.Valid };
+            dictionary["[1].IncludedProperty"] = new ModelState { ValidationState = ModelValidationState.Valid };
```

Comment:
There are two reasons why we should not use a model just to avoid creation of a new one. 
1. Maintainability & Readability
2. Maintainability & Readability
:smile: 

In this case I am confused reading the test case and wonder what is included/excluded, which is not the point of the test. Lets update this one.